packages/common/src/services/integrations/jwp/JWPAPIService.ts

```diff
@@ -0,0 +1,142 @@
+  withAuthentication?: boolean;
+  keepalive?: boolean;
+  contentType?: keyof typeof CONTENT_TYPES;
+  responseType?: 'json' | 'blob';
+  includeFullResponse?: boolean;
+};
+
+@injectable()
+export default class JWPAPIService {
+  private readonly storageService: StorageService;
+
+  private sandbox = true;
+
+  static isCommonError = (error: unknown): error is InPlayerError => {
+    return (
+      typeof error === 'object' &&
+      error !== null &&
+      'response' in error &&
+      typeof (error as InPlayerError).response?.data?.code === 'number' &&
+      typeof (error as InPlayerError).response?.data?.message === 'string'
+    );
+  };
+
+  constructor(@inject(StorageService) storageService: StorageService) {
+    this.storageService = storageService;
+  }
+
+  setToken = (token: string, refreshToken = '', expires: number) => {
+    return this.storageService.setItem(INPLAYER_TOKEN_KEY, JSON.stringify({ token, refreshToken, expires }), false);
+  };
+
+  getToken = async () => {
+    const tokenObject = await this.storageService.getItem(INPLAYER_TOKEN_KEY, true, false);
+
+    if (tokenObject) {
+      return tokenObject as { token: string; refreshToken: string; expires: number };
+    }
+
+    return { token: '', refreshToken: '', expires: 0 };
+  };
+
+  removeToken = async () => {
+    await Promise.all([this.storageService.removeItem(INPLAYER_TOKEN_KEY), this.storageService.removeItem(INPLAYER_IOT_KEY)]);
+  };
+
+  isAuthenticated = async () => {
+    const tokenObject = await this.getToken();
+
+    return !!tokenObject.token && tokenObject.expires > Date.now() / 1000;
+  };
+
+  private getBaseUrl = () => (this.sandbox ? 'https://staging-v2.inplayer.com' : 'https://services.inplayer.com');
+
+  private performRequest = async (
+    path: string = '/',
+    method = 'GET',
+    body?: Record<string, unknown>,
+    { contentType = 'form', responseType = 'json', withAuthentication = false, keepalive, includeFullResponse = false }: RequestOptions = {},
+    searchParams?: Record<string, string | number>,
+  ) => {
+    const headers: Record<string, string> = {
+      'Content-Type': CONTENT_TYPES[contentType],
+    };
+
+    if (withAuthentication) {
+      const tokenObject = await this.getToken();
+
+      if (tokenObject.token) {
+        headers.Authorization = `Bearer ${tokenObject.token}`;
+      }
+    }
+
+    const formData = new URLSearchParams();
+
+    if (body) {
+      Object.entries(body).forEach(([key, value]) => {
+        if (value || value === 0) {
+          if (typeof value === 'object') {
+            Object.entries(value as Record<string, string | number>).forEach(([innerKey, innerValue]) => {
+              formData.append(`${key}[${innerKey}]`, innerValue.toString());
+            });
+          } else {
+            formData.append(key, value.toString());
+          }
+        }
+      });
+    }
+
+    const endpoint = `${path.startsWith('http') ? path : `${this.getBaseUrl()}${path}`}${
```

Comment:
Do you think we also need to support `http` case here?

Reply:
Yes, we have one use case where we use an absolute path:
https://github.com/jwplayer/ott-web-app/pull/569/files#diff-9b06aa97ae60ba677a9a458cba1b082b563c63be0315d61b4a5ccd7f4ae535deR259